doc/source/serve/autoscaling-guide.md

```diff
@@ -0,0 +1,369 @@
+
+
+### [Optional] Define how the System reacts to changing traffic
+
+Given a steady stream of traffic and appropriately configured `min_replicas` and `max_replicas`, the steady state of your system is essentially fixed for a chosen configuration value for `target_num_ongoing_requests_per_replica`. Before reaching steady state, however, your system will be reacting to traffic shifts. How you want your system to react to changes in traffic determines how you want to set the remaining autoscaling configurations. We will go over two of the most commonly used parameters here.
+
+* **`upscale_delay_s [default=30s]`**: This defines how long Serve waits before scaling up the number of replicas in your deployment. More specifically, if the replicas are *consistently* serving more requests than desired for an `upscale_delay_s` number of seconds, then Serve will scale up the number of replicas based on aggregated ongoing requests metrics.
+* **`downscale_delay_s [default=600s]`**: This defines how long Serve waits before scaling down the number of replicas in your deployment. More specifically, if the replicas are *consistently* serving less requests than desired for a `downscale_delay_s` number of seconds, then Serve will scale down the number of replicas based on aggregated ongoing requests metrics.
+
+
+## Basic Example
+
+Let's go through an example synchronous workload that runs Resnet50. This is the application code:
+```{literalinclude} doc_code/resnet50_example.py
+:language: python
+:start-after: __serve_example_begin__
+:end-before: __serve_example_end__
+```
+
+This is the autoscaling configuration. Note that `target_num_ongoing_requests_per_replica` is set to 1 because the workload is synchronous. Also, the deployment starts with 0 replicas.
+```yaml
+applications:
+  - name: default
+    import_path: resnet:app
+    deployments:
+    - name: Model
+      max_concurrent_queries: 10
+      autoscaling_config:
+        target_num_ongoing_requests_per_replica: 1
+        min_replicas: 0
+        initial_replicas: 0
+        max_replicas: 200
+        upscale_delay_s: 30
+        downscale_delay_s: 600
+```
+
+We use [Locust](https://locust.io/) to run a load test against this application. The Locust load test will run a certain number of "users" that ping the Resnet50 service, where each user has a [constant wait time](https://docs.locust.io/en/stable/writing-a-locustfile.html#wait-time-attribute) of 0. This means each user will (repeatedly) send a request, wait for a response, then immediately send the next request. The number of users running over time is shown in the following graph:
+
+![users](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/resnet50_users.png)
+
+The results of the load test are as follows:
+
+::::{tab-set}
+:::{tab-item} Replicas
+<img src="https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling-guide/resnet50_replicas.png" alt="replicas" width="400"/>
```

Comment:
Is there any way to make the graphs a bit bigger? It's hard to see the axes.

Reply:
I've made the images bigger!